### PR TITLE
creating a form editing a user's profile creating a service request t…

### PR DESCRIPTION
**Pull request by DmitriyUshkvarok**
…o the server for editing a user saving the token to the global stack creating the functionality of component substitutability during editing actions validation of the form revalidation of data changes on the server loader when loading user's data.

### CODE REVIEW

File: src/app/profile/_components/FilledProfile/FiledProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          Profile image
+        </div>
+        {isSuccessEditUser && !isEditing ? (
+          <>
+            <ImageUpload onUpload={handleImageUpload} loading={loadingImg} />
+            <ProfileInfo
+              userData={userData}
+              isLoading={isLoading}
+              onEditClick={handleEditClick}
+              formatDate={formatDate}
```
Comment (Wdyffs) [inline src/app/profile/_components/FilledProfile/FiledProfile.tsx:120]: U do not need to pass formDate function as a prop, just use it as a separete utility function in the components

File: src/app/profile/_components/ProfileInfo/ProfileInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          <h2 className="mb-8 text-2xl font-medium text-primary">
+            Personal details
+          </h2>
+          <ul className="mb-10 flex flex-col gap-y-6">
+            <li className="flex">
+              <span className="w-[152px] text-primary opacity-60">
+                First name:
+              </span>
+              {userData.firstName}
+            </li>
```
Comment (Wdyffs) [inline src/app/profile/_components/ProfileInfo/ProfileInfo.tsx:31]: This should be a separate component to avoid code duplication

**Review by Wdyffs — CHANGES_REQUESTED** [review]

File: src/app/profile/_components/FilledProfile/FiledProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            initialUserData={userData || {}}
+          />
+        )}
+        <div className="mt-[51px]">
+          <h3 className="mb-[16px] text-2xl font-medium text-primary">
+            Password
+          </h3>
+          <Button className="flex items-center justify-center rounded-[47px] bg-secondary px-6 py-4 text-lg font-medium text-primary transition-opacity hover:opacity-60">
+            <span>Change password</span>
+          </Button>
```
[the remark this answers is not in the document]
Comment (DmitriyUshkvarok) [reply src/app/profile/_components/FilledProfile/FiledProfile.tsx:136]: fix

File: src/app/profile/_components/FilledProfile/FiledProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    const date = new Date(dateString)
+    const day = date.getDate()
+    const month = new Intl.DateTimeFormat('en-US', { month: 'long' }).format(
+      date,
+    )
+    const year = date.getFullYear()
+
+    return `${day} ${month} ${year}`
+  }
```
[the remark this answers is not in the document]
Comment (DmitriyUshkvarok) [reply src/app/profile/_components/FilledProfile/FiledProfile.tsx:84]: fix

File: src/app/profile/_components/FilledProfile/FiledProfile.tsx (hunk 1 above)
Comment (DmitriyUshkvarok) [reply src/app/profile/_components/FilledProfile/FiledProfile.tsx:120]: fix

File: src/app/profile/_components/ProfileInfo/ProfileInfo.tsx (hunk 2 above)
Comment (DmitriyUshkvarok) [reply src/app/profile/_components/ProfileInfo/ProfileInfo.tsx:31]: ?

File: src/app/profile/_components/FilledProfile/FiledProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+      try {
+        setIsLoading(true)
+
+        if (token) {
+          const userData = await getUserData(token)
+
+          setUserData(userData)
+        }
+      } catch (error) {
+        showError(error)
```
[the remark this answers is not in the document]
Comment (DmitriyUshkvarok) [reply src/app/profile/_components/FilledProfile/FiledProfile.tsx:49]: fix

File: src/app/profile/_components/ProfileInfo/ProfileInfo.tsx (hunk 2 above)
Comment (DmitriyUshkvarok) [reply src/app/profile/_components/ProfileInfo/ProfileInfo.tsx:31]: fix

File: src/utils/formDate.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    return 'N/A'
+  }
+
+  const date = new Date(dateString)
+  const day = date.getDate()
+  const month = new Intl.DateTimeFormat('en-US', { month: 'long' }).format(date)
+  const year = date.getFullYear()
+
+  return `${day} ${month} ${year}`
+}
```
Comment (horprogs) [inline src/utils/formDate.ts:13]: probably it's worth to use `date-fns` for all kind of date manipulation

**Review by Sunagatov — APPROVED** [review]
Approved

**Review by Sunagatov — APPROVED** [review]
Approved

**Pull request merged**